netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1717,9 +1759,27 @@ net_ebpf_extension_sock_addr_authorize_connection_classify(
         goto Exit;
     }
 
+    // First, try to find and use existing connection context from redirect layer
     verdict = _net_ebpf_ext_find_and_remove_connection_context(
         incoming_metadata_values->transportEndpointHandle, sock_addr_ctx);
 
+    if (verdict != BPF_SOCK_ADDR_VERDICT_REJECT) {
+        // No context found from redirect, invoke the BPF program for AUTH_CONNECT
+        // Set up original context for comparison (needed by verdict processing)
+        bpf_sock_addr_t sock_addr_ctx_original;
+        memcpy(&sock_addr_ctx_original, sock_addr_ctx, sizeof(sock_addr_ctx_original));
+        net_ebpf_sock_addr_ctx.original_context = &sock_addr_ctx_original;
+
+        program_result = net_ebpf_extension_hook_expand_stack_and_invoke_filtered_programs(
+            sock_addr_ctx, &filter_context->base, &verdict, _net_ebpf_extension_sock_addr_is_auth_connect_program);
```

Comment:
do we want to have any sort of arbitration here? For example, what happens if the connect (redirect) program issues a PROCEED_HARD, and then an auth_connect program issues a PROCEED_SOFT?

Are we just saying that if any bpf program has rejected at connect, then that's final. And then if no one rejects at connect layer, than auth_connect will be invoked, and we'll return the verdict there (regardless of the verdict at connect)?

Reply:
the WESP altitude model is "first rule match wins". what is the eBPF model if there are >1 programs attached to the same point? does it invoke all of them? or stop at the first one if the first one says BLOCK?